### DISCUSSION TITLE
The name of a page or blog or product in Cyrillic, is it possible to lay out automatic transliteration of the link?

### DISCUSSION THREAD

**Discussion by aukc1970** [givanz/Vvveb#210] [General]
Good day! Very good SMS! Just discovered when you create a new page or post, if you type the name of the page in Cyrillic, the link does not appear automatically, you need to manually type it in English. Is it possible to do it like in WordPress or Getsimple, when you type the name of the page in Cyrillic, the transliteration of the link name is automatically done? Thanks in advance for your answer! Translit is the writing of Cyrillic words in English letters.

**Comment by givanz** [comment]
Hi

Thank you for the suggestion, I bundled `Transliterate slugs and urls` plugin in the `lastest.zip` update.
You need to redownload the zip and update the files on your server then activate the plugin.

**Reply by aukc1970** [reply]
Thank you very much!

**Reply by aukc1970** [reply]
Good day! I made a new installation, activated the Transliterate slugs and urls plugin, but for some reason when you create a post or a new page and type the page name in Russian, for example, the page name Университет, the link is not automatically printed and not created, I did not see any changes. Maybe I'm doing something wrong?

![Screenshot_512](https://github.com/user-attachments/assets/a52cfd9a-902e-49c8-841b-d1dce6bd7d34)

**Reply by aukc1970** [reply]
I also wanted to clarify, for example, when you create a new page and type the name in English, the link is automatically printed and created, but when you type the page name in Cyrillic, for example in Russian, the link is not automatically printed and created, it must be typed manually. Is it possible to make it so that when you type the page name in Cyrillic, the link is automatically printed in transliteration in English.

**Reply by givanz** [reply]
One possible cause can be that the newly changed js file is not loaded.

Please make sure that `vvveb/public/admin/default/js/content.js` is updated/overwritten and clear Vvveb and browser cache.

You can also try to manually clear cache by deleting all files inside `/vvveb/vvveb/storage/compiled-templates/`

**Reply by aukc1970** [reply]
I did everything as you advised, but it didn't help. Nothing has changed yet, it doesn't work for me. Please describe how this plugin works? When you create a new blog or a new page and type the page name in Cyrillic, should the link be printed automatically or in some other way? Thank you very much for your help!

![Screenshot_512](https://github.com/user-attachments/assets/2466c70b-a6e7-434c-aaa6-94bb3b198028)
Here in the screenshot you can see that in the place where the field is circled in red, nothing was automatically printed and a hint was displayed that this field needs to be filled in.

**Reply by givanz** [reply]
Can you try on the demo? https://demo.vvveb.com/admin/index.php?module=content/post&post_id=6&type=post

The demo runs the same code as `latest.zip` downloaded from https://www.vvveb.com/

I recorded a short video on how it works on the demo.

https://github.com/user-attachments/assets/f1c8ad92-5a7b-4235-a074-88f252a6414c

Can you also check your browser console log (F12 key) to see if there are any javascript errors? 

This is probably caused by new js code not being loaded, probably because of cache.

**Comment by Yura-baza** [comment]
![Снимок экрана от 2024-11-05 12 32 26](https://github.com/user-attachments/assets/c8a38b0c-e136-483e-8412-b3304b63f8d2)
Не работает

**Reply by aukc1970** [reply]
Editing a page or post, in English the slug will be generated automatically from the name, but in Cyrillic the slug is not generated automatically from the name, please check, is it just me or you too?

![Screenshot_512](https://github.com/user-attachments/assets/a7277c53-b92a-4023-8b66-57e236add633)

**Reply by givanz** [reply]
Thanks for the bug report, I'm unable to reproduce the fatal error but I added code to prevent it, changed file `vvveb/plugins/transliterate/plugin.php` you need to redownload `latest.zip` and update the file.

**Comment by Yura-baza** [comment]
![Снимок экрана от 2024-11-05 20 14 47](https://github.com/user-attachments/assets/919f9380-0c56-478c-adb7-3b1a8bb9f148)
подправил немного вроде заработало

**Reply by givanz** [reply]
Thanks for feedback.

**Reply by Yura-baza** [reply]
Скачал с официального сайта переустановил движок все заработало, спасибо!

**Comment by Yura-baza** [comment]
![Снимок экрана от 2024-11-05 20 57 42](https://github.com/user-attachments/assets/46fd6dea-7a1c-4458-8405-e28ae9770a5a)

**Reply by aukc1970** [reply]
Thank you very much for your help! I tried it on your demo on my browser, apparently it's because of my browser or some extensions in my browser. I'll try it tomorrow on a clean browser, thank you very much, great CMS.

**Reply by aukc1970** [reply]
Thank you very much, I tried it on another browser and everything works.

**Reply by givanz** [reply]
Thanks for feedback.
